Adding a systems diagram to Armada. (#1885)
G-Research/armada: docs/design/diagrams/README.md

# Diagrams of Armada Architecture

## Systems Diagram

![Systems Diagram](armada-system.png)

This diagram shows the high-level architecture of Armada. It is made of 4 parts:

### Armada Server Components

> These are the components that control the operation of the different executors across different k8s clusters (See https://armadaproject.io/design)

Reviewer: Not sure what this means.

It includes:

> - **Armada Server**: The main component of Armada. It is responsible for the scheduling of jobs and the management of the executors.

Reviewer: Not sure if Armada Server is responsible for the management of the executors.  It can communicate with them but management means to me orchestrating.  I think the executors are independent components on kubernetes clusters and they can communicate to the server.

- **Lookout API**: A component that servers the lookout API, which enables users to monitor the status of jobs and queues.
- **Ingesters**: These move data from pulsar to their respective databases. The Event Ingester moves data from pulsar to redis. The Lookout Ingester moves data from pulsar to postgres for the lookout component to use.

### Armada Executor Components

These are the components that run on each k8s cluster and are responsible for the execution of jobs on that cluster.

> It includes:

Reviewer: Is this true?  Binoculars is stand alone from the executor?  It is built separately from executor.

- **Armada Executor**: The main component of the executor. It is responsible for the execution of jobs on the cluster.

> - **Binoculars**: A component that reads from the k8s API to get information about the cluster that can be used by Lookout.

Reviewer: ```suggestion
- **Binoculars**: A component that reads logs from the k8s API.
```

> # Diagrams of Armada Architecture

Author: This README needs redone for the new diagram.
